core/jvm/src/internal/SerializedValue.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2019-2024 JetBrains s.r.o. and contributors.
+ * Use of this source code is governed by the Apache 2.0 License that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.datetime.internal
+
+import kotlinx.datetime.*
+import java.io.*
+
+internal class SerializedValue(var typeTag: Int, var value: Any?) : Externalizable {
```

Comment:
Alternatively, we can place it in `kotlinx.datetime` package and name it shortly, e.g. `Ser`, to shave off some bytes.

Reply:
Do you mean, shave off some bytes from the serialized representation? I'm okay with that, as long as it still stays in the `internal/` directory.

Does it make sense to mark this with `@PublishedApi` to signal that it's an incompatible change to move/rename this class?

Reply:
> Does it make sense to mark this with @PublishedApi to signal that it's an incompatible change to move/rename this class?

Yes, I can do that